### PR TITLE
ipalib/frontend.py output_for_cli loops optimization

### PR DESCRIPTION
**Pull request by slaykovsky**
Trivial fix which removes unnecessary for loops.

### CODE REVIEW

File: ipalib/frontend.py [hunk 1]
```diff
@@ -1041,18 +1043,23 @@ def output_for_cli(self, textui, output, *args, **options):
 
         self.log_messages(output)
 
-        order = [p.name for p in self.output_params()]
+        order = deque()
+        labels = dict()
```
Comment (tiran) [inline ipalib/frontend.py]:
```
labels = {}
flags = {}
```

It's shorter, easier to read, and faster.

File: ipalib/frontend.py [hunk 2]
```diff
@@ -1041,18 +1043,23 @@ def output_for_cli(self, textui, output, *args, **options):
...
+        flags = dict()
+
+        for p in self.output_params():
+            order.append(p.name)
+            labels[p.name] = unicode(p.label)
+            flags[p.name] = p.flags
+
         if options.get('all', False):
-            order.insert(0, 'dn')
+            order.appendleft('dn')
```
Comment (tiran) [inline ipalib/frontend.py]: Unless it's performed in a loop, one ``insert(0, item)`` is ok. A deque is overkill. It requires more memory and doesn't amortize that fast.

File: ipalib/frontend.py (hunk 1 above)
Comment (slaykovsky) [reply ipalib/frontend.py]: Okay.

File: ipalib/frontend.py (hunk 2 above)
Comment (slaykovsky) [reply ipalib/frontend.py]: Will return list then :)

**Comment by tiran** [comment]
Good catch! 🥇

**Comment by tomaskrizek** [comment]
Please rebase to fix the testing jobs, it's caused by switching to F27 template.

**Comment by felipevolpone** [comment]
@slaykovsky could you, please, rebase this PR to PR CI get it right?

**Comment by tiran** [comment]
master:

* 197b5ca639ab31bc5f079c45eb36763c442bf74c ipalib/frontend.py output_for_cli loops optimization

**Pull request closed without merging**